src/test/java/org/stellar/sdk/MemoTest.java

```diff
@@ -122,18 +144,21 @@ public void testMemoHashInvalidHex() {
         try {
             Memo.hash("test");
             fail();
-        } catch (Exception e) {
-
+        } catch (MemoWrongSizeException e1) {
+            fail();
+        } catch (IllegalArgumentException e2) {
```

Comment:
It's not immediately clear where this IAE is thrown from? I kinda expected MemoWrongSizeException in this case?

Reply:
I thought the same at first but since we are parsing the string to decode it to bytes, it makes sense to have the IAE.

Reply:
I think the specialized Memo exceptions are unnecessary granularity for clients to juggle, the excepts are un-checked, `RuntimeException`, which means callers can't recover from different types of this error and try something else, if that was the case then the Memo exceptions should extend `Exception` instead and be thrown from methods as checked, forcing callers to catch that, but since that is not applicable here, their granularity can just be expressed as different messages using the standard `RuntimeException`.

I would suggest to remove MemoWrongSizeException and MemoTooLongException , and just throw `new RuntimeException("memo problem explained here")` where needed, then this test just verifies `catch (Exception)` across all bad inputs, callers can't do something different depending on the type of invalid memo here, they have to move on and the msg provides context for logging purposes.

Thoughts otherwise?

There's a minor breaking change there on removing existing MemoTooLongException, I think it's worthwhile as it's less code for clients to manage, we have to mention it in the CHANGELOG which we do at release prep time, but if you want to add a note under new `pending` section at top of CHANGELOG.md, we can roll that in during the next release notes.